smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/providers/HttpQueryItemProvider.kt

```diff
@@ -141,6 +142,15 @@ class HttpQueryItemProvider(
                     renderQueryItem(queryBinding.member, bindingIndex, memberName, paramName)
                 }
             }
+            if (queryBinding.member.isRequired()) {
+                writer.openBlock("else {", "}") {
+                    writer.write(
+                        "let message = \"Creating a URL Query Item failed. \$L is required but it is nil\"",
+                        memberName
+                    )
+                    writer.write("throw ClientRuntime.ClientError.queryItemCreationFailed(message)")
+                }
+            }
```

Comment:
we should use a guard let than if let here to go with Swift style when property is required